hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/TransactionHandler.java

```diff
@@ -33,6 +34,18 @@ public interface TransactionHandler {
      */
     void preHandle(@NonNull final PreHandleContext context) throws PreCheckException;
 
+    /**
+     * Validate the transaction body, without involving state or dynamic properties.
+     * This method is called as first step of preHandle. If there is any failure,
+     * throws a {@link PreCheckException}.
+     * Since these checks are pure, they need not be repeated in handle workflow.
+     * The result of these checks is cached in the {@link PreHandleContext} for use
+     * in handle workflow.
+     * @param txn the transaction body
+     * @throws PreCheckException if the transaction is invalid
+     */
+    void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException;
```

Comment:
I think we should have an empty default method.

Reply:
You think that is needed if any handler has no checks to do ?

Reply:
What check is required by every handler?
Also, I second the request for a default so we can minimize the amount of merge conflicts this will trigger (particularly for those who already have a pureChecks method in progress on a branch).

Reply:
Any validation that are done in `preHandle` based on just the `TransactionBody` and not involving state or config is needed by any handler as far as I know.

Reply:
I don't know for certain that every preHandle has checks that meet the criteria.  I get that any that do will move the checks to pureChecks, just noting that it's possible some won't have any.

Reply:
Adding the methods in every handler forces merge conflicts with any service that already has a pureChecks method.  That's why I suggested removing them.  If we're keeping the implementations everywhere, then not much point to a default version.

Reply:
Added FUTURE task to remove default method. Fixed as suggested to void merge conflicts